base/ltfssaxes.dtx

```diff
@@ -280,23 +570,69 @@
 \DeclareFontSeriesChangeRule {lc}{b}{bc}{}
 \DeclareFontSeriesChangeRule {lc}{eb}{ebc}{}
 \DeclareFontSeriesChangeRule {lc}{ub}{ubc}{}
+\DeclareFontSeriesChangeRule {lc}{uc}{luc}{}
 \DeclareFontSeriesChangeRule {lc}{ec}{lec}{}
 \DeclareFontSeriesChangeRule {lc}{c}{lc}{}
 \DeclareFontSeriesChangeRule {lc}{sc}{lsc}{}
+\DeclareFontSeriesChangeRule {lc}{sx}{lsx}{}
 \DeclareFontSeriesChangeRule {lc}{x}{lx}{}
+\DeclareFontSeriesChangeRule {lc}{ex}{lex}{}
+\DeclareFontSeriesChangeRule {lc}{ux}{lux}{}
 %    \end{macrocode}
 %
 %    \begin{macrocode}
-\DeclareFontSeriesChangeRule {l}{sb}{sb}  {b}   % ?         %<-----
-\DeclareFontSeriesChangeRule {l}{b}{b}  {bx}                %<-----
+\DeclareFontSeriesChangeRule {lsc}{ul}{ulsc}{}
+\DeclareFontSeriesChangeRule {lsc}{el}{elsc}{}
+\DeclareFontSeriesChangeRule {lsc}{l}{lsc}{}
+\DeclareFontSeriesChangeRule {lsc}{sl}{slsc}{}
+\DeclareFontSeriesChangeRule {lsc}{sb}{sbsc}{}
+\DeclareFontSeriesChangeRule {lsc}{b}{bsc}{}
+\DeclareFontSeriesChangeRule {lsc}{eb}{ebsc}{}
+\DeclareFontSeriesChangeRule {lsc}{ub}{ubsc}{}
+\DeclareFontSeriesChangeRule {lsc}{uc}{luc}{}
+\DeclareFontSeriesChangeRule {lsc}{ec}{lec}{}
+\DeclareFontSeriesChangeRule {lsc}{c}{lc}{}
+\DeclareFontSeriesChangeRule {lsc}{sc}{lsc}{}
+\DeclareFontSeriesChangeRule {lsc}{sx}{lsx}{}
+\DeclareFontSeriesChangeRule {lsc}{x}{lx}{}
+\DeclareFontSeriesChangeRule {lsc}{ex}{lex}{}
+\DeclareFontSeriesChangeRule {lsc}{ux}{lux}{}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\DeclareFontSeriesChangeRule {l}{uc}{luc}  {l}   % ?        %<-----
 \DeclareFontSeriesChangeRule {l}{ec}{lec}  {l}   % ?        %<-----
 \DeclareFontSeriesChangeRule {l}{c}{lc}  {l}   % ?          %<-----
 \DeclareFontSeriesChangeRule {l}{sc}{lsc}  {l}   % ?        %<-----
+\DeclareFontSeriesChangeRule {l}{sx}{lsx}  {l}   % ?        %<-----
 \DeclareFontSeriesChangeRule {l}{x}{lx}  {l}   % ?          %<-----
+\DeclareFontSeriesChangeRule {l}{ex}{lex}  {l}   % ?        %<-----
+\DeclareFontSeriesChangeRule {l}{ux}{lux}  {l}   % ?        %<-----
+\DeclareFontSeriesChangeRule {l}{sb}{sb}  {b}   % ?         %<-----
+\DeclareFontSeriesChangeRule {l}{b}{b}  {bx}                %<-----
 \DeclareFontSeriesChangeRule {l}{bx}{bx}  {b}               %<-----
```

Comment:
There are a number of entries of the form `\DeclareFontSeriesChangeRule{...}{...}{bx}{b}`. I suspect that they exist because for the Computer Modern fonts 'bold' is `bx` and not `b`, so that you have a reasonable substitution when `bx` is applied to another font. Wouldn't it be better if this was handled in a general way, e.g., in `\merge@font@series@`, instead of handling this case (incompletely) via random `\DeclareFontSeriesChangeRule` entries?

Reply:
It would not, I think, because the rules are adaptable, so if oyu have a special job that needs changes in that respect you can overwrite the rule. If it would be done internally your hand is forced.

But in the opposite direction this means that there is no point in adding such alternatives in situations where the extended shapes play no role, e.g., if ultra light is an existing series then we have a font like noto and then falling back to bx or b doesn't make much sense

Reply:
So you think it would be better to delete the entries of the form `\DeclareFontSeriesChangeRule{...}{...}{bx}{b}` (`\DeclareFontSeriesChangeRule{...}{...}{b}{bx}`) altogether?

Reply:
I have mixed feelings about the b / bx. I think it is probably helpful to have the rules that ask for `bx` fall back to `b` because the `bx`might just come from a wrong `\bfseriesdefault`.  Similar, if CM or LM fonts + some others are used, then the setting might have changed to `b`and so fails with CM or LM. Thus, I think keeping those makes some sense. The situation is different if we are already in some series that is clearly outside the main ones, then they are much less useful, though they don't hurt.